Source/Common/singleton.cpp

```diff
@@ -30,33 +29,59 @@ http_singleton::http_singleton()
 
 http_singleton::~http_singleton()
 {
+    for (auto& mockCall : m_mocks)
```

Comment:
Maybe should just call HCMockClearMocks()

Reply:
It seems safer to do the cleanup on the object being destroyed vs calling ClearMocks which cleans up the global again.

Reply:
Yes, I'd just like to avoid duplicating the code, maybe a helper function on the singleton that's called from both places?